### PR TITLE
Update js documentation for shop parameters pages

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |1.7.8.x
| Description?      | Update js documentation for shop parameters pages
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no need
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24551)

### CODE REVIEW

File: tests/UI/pages/BO/shopParameters/contact/add.js [hunk 1]
```diff
@@ -43,8 +52,8 @@ class AddContact extends BOBasePage {
 
   /**
    * Fill form for add/edit contact
-   * @param page
-   * @param contactData
+   * @param page {Page} Browser tab
+   * @param contactData {object} Data to set on contact form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/contact/add.js]:
```suggestion
   * @param contactData {ContactData} Data to set on contact form
```

Maybe we can link it after that if we add js-doc on faker data files

File: tests/UI/pages/BO/shopParameters/customerSettings/titles/add.js [hunk 2]
```diff
@@ -48,8 +57,8 @@ class AddTitle extends BOBasePage {
 
   /**
    * Fill title form and get successful message
-   * @param page
-   * @param titleData
+   * @param page {Page} Browser tab
+   * @param titleData {object} Data to set on create/edit title form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/customerSettings/titles/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/add.js [hunk 3]
```diff
@@ -28,8 +37,8 @@ class AddOrderStatus extends BOBasePage {
 
   /**
    * Fill order status form in create or edit page and save
-   * @param page
-   * @param orderStatusData
+   * @param page {Page} Browser tab
+   * @param orderStatusData {object} Data to set on order status form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/returnStatus/add.js [hunk 4]
```diff
@@ -17,9 +26,9 @@ class AddOrderReturnStatus extends BOBasePage {
   /* Methods */
 
   /**
-   * Fill order return status form in create or edit page and save
-   * @param page
-   * @param orderReturnStatusData
+   * Fill order return status form
+   * @param page {Page} Browser tab
+   * @param orderReturnStatusData {object} Data to set on order return status form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/returnStatus/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/search/add.js [hunk 5]
```diff
@@ -17,8 +26,8 @@ class AddAlias extends BOBasePage {
   /* Methods */
   /**
    * Create/Edit alias
-   * @param page
-   * @param aliasData
+   * @param page {Page} Browser tab
+   * @param aliasData {object} Data to set on alias form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/search/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/search/tags/add.js [hunk 6]
```diff
@@ -20,8 +29,8 @@ class AddTag extends BOBasePage {
   /* Methods */
   /**
    * Create/Edit tag
-   * @param page
-   * @param tagData
+   * @param page {Page} Browser tab
+   * @param tagData {object} Data to set to tag form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/search/tags/add.js]: same here

File: tests/UI/pages/BO/shopParameters/stores/add.js [hunk 7]
```diff
@@ -32,8 +41,8 @@ class AddStore extends BOBasePage {
 
   /**
    * Fill creation / edition form for store and save it
-   * @param page
-   * @param storeData
+   * @param page {Page} Browser tab
+   * @param storeData {object} Data to set on store form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/stores/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/stores/index.js [hunk 8]
```diff
@@ -413,9 +421,9 @@ class Stores extends BOBasePage {
   /* Form functions */
   /**
    * Se contact details
-   * @param page
-   * @param storeContactData
-   * @returns {Promise<unknown>}
+   * @param page {Page} Browser tab
+   * @param storeContactData {object} Store contact data to set on contact detail form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/stores/index.js]: Same here

File: tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/add.js [hunk 9]
```diff
@@ -21,8 +30,8 @@ class AddSearchEngine extends BOBasePage {
 
   /**
    * Fill create or edit search engine form and save it
-   * @param page
-   * @param searchEngineData
+   * @param page {Page} Browser tab
+   * @param searchEngineData {object} Data to set on search engine form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/trafficAndSeo/seoAndUrls/add.js [hunk 10]
```diff
@@ -73,8 +81,8 @@ class AddSeoUrl extends BOBasePage {
 
   /**
    * Create/Edit seo page
-   * @param page
-   * @param seoPageData
+   * @param page {Page} Browser tab
+   * @param seoPageData {object} Data to set on seo form
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/shopParameters/trafficAndSeo/seoAndUrls/add.js]: Same here

File: tests/UI/pages/BO/shopParameters/contact/add.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/shopParameters/contact/add.js]: Done tx :+1:

**Review by NeOMakinG — DISMISSED** [review]

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js [hunk 11]
```diff
@@ -141,11 +147,11 @@ class Statuses extends BOBasePage {
...
-   * @param page
-   * @param tableName
-   * @param filterType
-   * @param filterBy
-   * @param value
+   * @param page {Page} Browser tab
+   * @param tableName {string} Table name to filter
+   * @param filterType {string} Type of filter (input/select)
+   * @param filterBy {string} Column to filter with
+   * @param  value {string} value to filter with
```
Comment (Progi1984) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]:
```suggestion
   * @param value {string} value to filter with
```

**Review by Progi1984 — COMMENTED** [review]
A last comment

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js (hunk 11 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/shopParameters/orderSettings/statuses/index.js]: Thx done!

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**